### ISSUE TITLE
Buffer is not filled as intended for Dash mpd with multiple periods

### ISSUE THREAD

**Issue by juechemparathy** [google/ExoPlayer#5182]
https://gist.github.com/juechemparathy/ce40519847c5d4bc3451d5b5b680f5fc
Above mpd has each segment as ~4s periods and I see that at any point only 4s of buffer is loaded instead of accumulating till the HighWaterMark level set on LoadController. 

Exoplayer version: 2.9.0
Stream info:
 ```
     {
        "name": "LIVE - Multiple Periods",
        "uri": "https://content.uplynk.com/channel/3c367669a83b4cdab20cceefac253684.mpd?ad=cleardashnew",
        "drm_scheme": "widevine",
        "drm_multi_session": true,
        "drm_license_url": "https://content.uplynk.com/wv",
        "drm_key_request_properties": {
          "X-DASH-SEND-ALL-KEYS": "1"
        },
        "extension": "mpd"
      } 
```
Live stream has 3 minutes of slate and 3 minutes of DRM content rolling.
Slate is represented as mulitple periods of 4s.

Red box indicate buffer state during the slate.
![screen shot 2018-11-30 at 12 48 08 pm](https://user-images.githubusercontent.com/3212941/49337626-f0fb4c80-f5ca-11e8-8836-8372fae2eab5.png)

**Comment by ojw28** [comment]
I think the buffer probably is being filled as intended, but that the position is being reported incorrectly. I'm also fairly sure it was fixed in 2.9.1. The corresponding line in the release notes is:

> Fix issue where the buffered position was not updated correctly when transitioning between periods (#4899).

If you still see this behavior in 2.9.1 or later, please let us know! Marking as a duplicate of #4899

**Comment by juechemparathy** [comment]
LGTM with 2.9.1. Thanks.
